src/adapters/communities-db.ts

```diff
@@ -267,9 +269,17 @@ export function createCommunitiesDBComponent(
         WHERE c.active = true
       `
 
-      const membersJoin = options.onlyMemberOf
-        ? SQL` JOIN community_members cm ON c.id = cm.community_id AND cm.member_address = ${normalizedMemberAddress}`
-        : SQL` LEFT JOIN community_members cm ON c.id = cm.community_id AND cm.member_address = ${normalizedMemberAddress}`
+      let membersJoin = SQL``
+
+      if (onlyMemberOf && roles) {
+        membersJoin = SQL` JOIN community_members cm ON c.id = cm.community_id AND cm.member_address = ${normalizedMemberAddress} AND cm.role = ANY(${roles})`
+      } else if (onlyMemberOf) {
+        membersJoin = SQL` JOIN community_members cm ON c.id = cm.community_id AND cm.member_address = ${normalizedMemberAddress}`
+      } else if (roles) {
+        membersJoin = SQL` JOIN community_members cm ON c.id = cm.community_id AND cm.member_address = ${normalizedMemberAddress} AND cm.role = ANY(${roles})`
+      } else {
+        membersJoin = SQL` LEFT JOIN community_members cm ON c.id = cm.community_id AND cm.member_address = ${normalizedMemberAddress}`
+      }
```

Comment:
Do you think we could move this to the `queries.ts` file and possibly reduce the number of' if' statements in the new function so we write the least SQL possible?

I imagine something like:

```js
const baseJoin = SQL` JOIN community_members cm ON c.id = cm.community_id AND cm.member_address = ${normalizedMemberAddress}`

if (roles && roles.length > 0) {
  return baseJoin.append(SQL` AND cm.role = ANY(${roles})`)
}

if (onlyMemberOf) {
  return baseJoin
}

return SQL` LEFT `.append(baseJoin)
```

We would be able then to reuse it for the `getCommunitiesCount`.